Fix `KeyError` issue when plotting

For some reason, when I run `test_plot_2d_figure_error`, the test sometimes fails with the following strange error:

```python
>       fp.plot(
            euclid_opt.marginalize_over("Omegam", "Omegab", invert=True),
        )

tests/test_graphics.py:783:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
fitk/graphics.py:2125: in plot
    ax[i, i].remove()
.venv/lib/python3.11/site-packages/matplotlib/artist.py:212: in remove
    self._remove_method(self)
.venv/lib/python3.11/site-packages/matplotlib/figure.py:922: in delaxes
    self._axstack.remove(ax)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <matplotlib.figure._AxesStack object at 0x7fbdf87b4390>, a = <AxesSubplot: >

    def remove(self, a):
        """Remove the axes from the stack."""
>       self._axes.pop(a)
E       KeyError: <AxesSubplot: >

.venv/lib/python3.11/site-packages/matplotlib/figure.py:75: KeyError
```

Note that I am not sure what the exact source of the error is, as it appears randomly when I run the tests.
I suppose one easy fix is to do something like:

```python
try:
    ax[i, i].remove()
except KeyError:
    pass
```

though I'm not entirely sure _why_ this is the case as I explicitly perform checks for this beforehand, so this module should never throw any exceptions.

--- a/fitk/graphics.py
+++ b/fitk/graphics.py
@@ -2048,7 +2048,10 @@ class FisherFigure2D(_FisherMultipleAxesFigure):
                 if i < j and self.figure is None:
                     # calling `remove()` is better than `axis("off")` because
                     # then we obtain the proper bounding box
-                    ax[i, j].remove()
+                    try:
+                        ax[i, j].remove()
+                    except KeyError:
+                        pass
 
                 # plotting the 2D contours
                 elif i > j:
@@ -2146,7 +2149,10 @@ class FisherFigure2D(_FisherMultipleAxesFigure):
 
                     else:
                         if self.figure is None:
-                            ax[i, i].remove()
+                            try:
+                                ax[i, i].remove()
+                            except KeyError:
+                                pass
 
             # set automatic limits
             for i, j in product(range(size), repeat=2):

